smarts/env/wrappers/gif_recorder.py

```diff
@@ -0,0 +1,40 @@
+import os
+
+from moviepy.editor import *
+import gym.envs
+import shutil
+
+
+class GifRecorder:
+    def __init__(self, dir, env):
+        self.dir = dir
+        self.env = env
+
+        try:
+            os.mkdir(self.dir)
+        except:
+            pass
+
+        self.dir_name = None
+        if "/" not in dir:
+            self.dir_name = dir
+        else:
+            last_index = dir.rindex("/")
+            self.dir_name = dir[-(len(dir) - last_index - 1) :]
+
+    def capture_frame(self, step_num):
+        image = self.env.render(mode="rgb_array")
```

Comment:
There are three directions we could go with this, we can keep as is:

```suggestion
    def capture_frame(self, step_num):
        image = self.env.render(mode="rgb_array")
```

We can remove `self.env` and pass in the environment on render:

```suggestion
    # pass in env
    def capture_frame(self, step_num, env):
        image = env.render(mode="rgb_array")
```

or we could only pass in the image (my preference):

```suggestion
    # elsewhere call 
    # > `gif_recorder.capture_frame(
    # >     get_next_id(),
    # >     env.render(mode="rgb_array"),
    # > )  
    def capture_frame(self, step_num, image):
```